### PR TITLE
Modal component

### PR DESCRIPTION
**Pull request by viniciushvc**
It is only with the code part, with a11y, still missing the design

### CODE REVIEW

File: src/components/modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        </S.Content>
+      </DialogPrimitive.Portal>
+    </DialogPrimitive.Root>
+  );
+});
+
+// Type for onInteractOutside method
+type OverlayClickEvent = CustomEvent<{
+  originalEvent: PointerEvent | FocusEvent;
+}>;
```
Comment (codeferreira) [inline src/components/modal/index.tsx:91]: Why is this type on the end of the file instead of the top with other types?

**Review by codeferreira — COMMENTED** [review]
We just need to take a look at this small code organization issue

File: src/components/modal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <DialogPrimitive.Portal>
+        <S.Overlay />
+
+        <S.Content
+          ref={ref}
+          onInteractOutside={handleOverlayClick}
+          onCloseAutoFocus={handleModalClose}
+        >
+          <DialogPrimitive.Close asChild>
+            <S.IconButton>&times;</S.IconButton>
```
Comment (miguel-nascimento) [inline src/components/modal/index.tsx:78]: Need to change this later, but for now looks good

**Review by miguel-nascimento — APPROVED** [review]

File: src/components/modal/index.tsx (hunk 1 above)
Comment (miguel-nascimento) [reply src/components/modal/index.tsx:91]: This type is created just to ensure typescript that everything is okay, IMO no needed to put in the top of the file

**Comment by miguel-nascimento** [comment]
Merging for review

**Pull request merged**